### PR TITLE
Form Validation in Vue.js using VeeValidate

### LINKED ISSUE #3137: Form Validation in Vue.js using VeeValidate
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### Proposed title of article
Form Validation in Vue.js using Veevalidate

### Introduction paragraph (2-3 paragraphs):
Client-side validation is common in today's web applications and pages, it improves user experience and saves on application resources. In this article, we explore the VeeValidate library validating our form(experimenting on different fields), best practices of client-side validation, creating custom validation rules, and take a look at validation triggers.
I will also provide a tailwind form template for practice.

### Key takeaways: 
The reader will gain the following at the end of the article.
 - Understand the importance of client-side validation.
 - Understand the in's and out's of vee validate library.
 - Be able to validate your own forms.
 - Be able to create custom validation rules.
 - Understand and customize validation triggers.

### References: 
Please list links to any published content/research that you intend to use to support/guide this article.
https://vee-validate.logaretm.com/v4/

### Templates to use as guides

- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Njunu-sk**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**

This closes #3137.
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [X] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [X] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [X] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [X] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [X] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [X] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [X] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [X] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [X] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [X] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [X] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [X] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [X] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [X] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [X] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

File: content/articles/form-validation-in-vue.js-using-veevalidate/index.md [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+
+We are provided with the above components:
+
+- **Form** component, this renders a form, wraps all form elements.
+- **Field** component, represents form inputs and renders any HTML element.
+- **ErrorMessage** component, displays an error message for a field.
+
+Let's define the validation process and see how it relates to the mentioned components:
+
+1. Wrap your form with the **Form** component, overwriting the default _form_ elements.
```
Comment (katungi) [inline content/articles/form-validation-in-vue.js-using-veevalidate/index.md:62]: Fix the numbering issue here.

**Review by katungi — COMMENTED** [review]
Hey @Njunu-sk , Thanks for submitting a PR. I just started the preliminary review.
Just as a note: 
- Always specify the language of a code snippet. This helps in color formatting of the snippet in production.
- Avoid very long sentences. Some of the sentences are very long. The reader might get bored with the article.

Overall , the article is really good so far, just a few changes and we can do the tech review. Be sure to check on the changes I have suggested, and those I have made.

File: content/articles/form-validation-in-vue.js-using-veevalidate/index.md (hunk 1 above)
Comment (Njunu-sk) [reply content/articles/form-validation-in-vue.js-using-veevalidate/index.md:62]: hi, thanks for the comment but think the md is ok(https://guides.github.com/features/mastering-markdown/)

Comment (katungi) [reply content/articles/form-validation-in-vue.js-using-veevalidate/index.md:62]: Ah thanks. Then that makes sense. Thanks, I did not know about that!

**Comment by louisefindlay23** [comment]
@Njunu-sk, please remember to fill in the `This closes #issue-number` in the Article Checklist by removing the backticks and replacing `issue-number` with the issue number of your topic suggestion so you can be correctly paid out and your topic suggestion closed when this has been merged. Have done this for you above.

**Comment by katungi** [comment]
Hello @Njunu-sk , the article needs a hero image. Remember the image has to be named `hero` and should be `600 x 400`. You can use [pexels](https://www.pexels.com/) or our [community templates](https://github.com/ahmadmardeni1/Canva) to get one and if you have any other source, feel free to use it.

**Comment by Njunu-sk** [comment]
@katungi, thank you for the review and kind comments.

**Comment by hectorkambow** [comment]
@katungi Is this article ready for final review?? (blue label)??

**Comment by katungi** [comment]
Ha! thanks @hectorkambow , that was a silly mistake on my side.

**Comment by LinusMuema** [comment]
Hi @Njunu-sk 
This article has quite a lot of info and that is great. But I had to make a lot of changes to grammar, especially punctuation. Please check on the use of full stops and commas to check the difference. There was also an issue with links. Read more on markdown formatting to check how anchors are added. Also, the readability of the article could use some more work. Please check my commits for future articles.

cc: @katungi please check these issues during review so as to allow faster publishing of articles.

**Comment by hectorkambow** [comment]
@Njunu-sk 

Please be sure to go through the edits made by your peer reviewer and content moderator to improve future articles.

As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published.
Our team of peer reviewers do not provide revisions services at this time in order to review them all in a timely manner. 

All revisions and edits should be completed on your own forked repo (as to not take up room in the queue). 
Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

cc @katungi articles that require this level of review should be closed. To allow the student to make the revisions on their (and to open a new) pr at that moment. 

cc @LinusMuema

**Pull request merged**